### PR TITLE
Convert to noarch, remove link scripts

### PR DESCRIPTION
**Pull request by asford**
This is a strawman-PR that implements the changes in #52 via an incremented build number. This causes a risky upgrade path in some legacy notebook scenarios, documented in #52, and would likely be better executed via a version update or (perhaps) a custom channel for the 3.5.1 series?

See
https://github.com/conda-forge/widgetsnbextension-feedstock/issues/52

See
https://github.com/conda-forge/widgetsnbextension-feedstock/issues/58

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by asford** [comment]
@conda-forge-admin, please rerender

**Comment by jasongrout** [comment]
We're over a year past the upgrade timeline in #52, and have no immediate timeline for a version bump, so perhaps it is just as well to move forward with this in a build version bump. If anyone has problems with this change in a build version bump, please let us know.

We might have a small version bump (patch, maybe?) in conjunction with the jlab 3 update, but we're still working on that.

@ocefpaf - thanks for merging, though I would have appreciated a bit more time for the ipywidgets devs to comment, as there were some trickier upgrade nuances here since we had previously made some mistakes in packaging.

**Comment by ocefpaf** [comment]
> @ocefpaf - thanks for merging, though I would have appreciated a bit more time for the ipywidgets devs to comment, as there were some trickier upgrade nuances here since we had previously made some mistakes in packaging.

It worked in my tests. Let me know if you find any mistakes in this. Marking packages as broken is easy with PRs to https://github.com/conda-forge/admin-requests

**Comment by asford** [comment]
@ocefpaf @jasongrout Mea culpa! I should have marked this PR as a draft for context.

Perhaps we should just cut a quick build `4` to revert this change for the time being? This would maintain the behavior for the "average user" just running an update, which would presumably be the most likely to be affected by this weird legacy-update-in-standing-environment flow, but allow us to pin against the specific build number to workaround until a version bump.

**Comment by asford** [comment]
I've opened a quick revert of this in #62 with another build number bump if there are concerns about upgrade nuances.

**Comment by ocefpaf** [comment]
> I've opened a quick revert of this in #62 with another build number bump if there are concerns about upgrade nuances.

You'll also need to mark the packages as broken. Reverting won't remove them.

**Comment by jasongrout** [comment]
I think we can leave this as-is for now, and revert if there are problems that people are experiencing. The main reason to convert on a version boundary is to improve communication with people if there are issues, but since we are so far past the original upgrade issue, I don't anticipate issues.

Thanks for opening this pr @asford.

**Comment by asford** [comment]
@ocefpaf Correct me if I'm wrong, but the updated build number `4` will override the `3` build for unpinned updates correct? `3` isn't technically "broken", and is (in fact) superior for the "happy path", but could trigger a broken environment in a specific set of legacy edge cases. I'd really like to be able to specifically pin `3` in my environment if at all possible.

**Comment by asford** [comment]
Roger, thanks for the clarification @jasongrout 

I'm happy to "bow-out" here and defer to y'all on the ipython side, but would very much appreciate if build number `3` could remain available. Given that the use of the unlink script is relatively low impact except for a few specific scenarios my soft-vote would be toward cutting a build number `4` to restore the old behavior to eliminate the support risk.

**Comment by ocefpaf** [comment]
> @ocefpaf Correct me if I'm wrong, but the updated build number `4` will override the `3` build for unpinned updates correct? `3` isn't technically "broken", and is (in fact) superior for the "happy path", but could trigger a broken environment in a specific set of legacy edge cases. I'd really like to be able to specifically pin `3` in my environment if at all possible.

Conda should prioritize the higher bn but that is not always true. I know that mamba does though.
TL;DR the "right" way to rever is to remove the previous package. But I don't think it is necessary.

**Comment by jasongrout** [comment]
> my soft-vote would be toward cutting a build number `4` to restore the old behavior to eliminate the support risk.

On second thought, I agree. It's safer to do this on a version bump for people running old versions of the notebook or ipywidgets. +1 to leaving build 3 as-is for those that need a noarch package too.
